server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/ExportService.java

```diff
@@ -0,0 +1,94 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.vault.routes;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.apache.james.blob.api.BlobStore;
+import org.apache.james.blob.sharing.api.BlobSharingMechanism;
+import org.apache.james.core.MailAddress;
+import org.apache.james.core.User;
+import org.apache.james.vault.DeletedMessage;
+import org.apache.james.vault.DeletedMessageVault;
+import org.apache.james.vault.DeletedMessageWithContent;
+import org.apache.james.vault.DeletedMessageZipper;
+import org.apache.james.vault.search.Query;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+class ExportService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExportService.class);
+
+    private final BlobSharingMechanism blobSharing;
+    private final BlobStore blobStore;
+    private final DeletedMessageZipper zipper;
+    private final DeletedMessageVault vault;
+
+    ExportService(BlobSharingMechanism blobSharing, BlobStore blobStore, DeletedMessageZipper zipper, DeletedMessageVault vault) {
+        this.blobSharing = blobSharing;
+        this.blobStore = blobStore;
+        this.zipper = zipper;
+        this.vault = vault;
+    }
+
+    Mono<Void> export(User user, Query exportQuery, MailAddress shareeAddress) throws IOException {
+        Stream<DeletedMessageWithContent> fullMessages = Flux.from(vault.search(user, exportQuery)).toStream()
+            .map(deletedMessage -> messageWithContent(user, deletedMessage))
```

Comment:
Even I use Reactor chains for this stuff, there is still no benefit regards to the performance because what ever the way is, I am using a single thread from TaskManager executor. Then I went to the simple impl to not use Reactor if possible.

Reply:
Well don't you think we can parralelize a bit these IO?

Reply:
okay